src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -68,82 +66,52 @@ internal class RenameEndpoint : IRenameEndpoint
             return new RegistrationExtensionResult(ServerCapability, options);
         }
 
-        public async Task<WorkspaceEdit?> Handle(RenameParamsBridge request, CancellationToken cancellationToken)
-        {
-            if (request is null)
-            {
-                throw new ArgumentNullException(nameof(request));
-            }
-
-            if (!_languageServerFeatureOptions.SupportsFileManipulation)
```

Comment:
Current design is lacking this check. Debating how we fit this in... basically it's an implementation saying to exit early based on some check. Seem unique to rename , but I wonder if we should add an `virtual bool IsSupported()` that would allow implementations to exit early based on configuration. 

@davidwengier @NTaylorMullen thoughts?

Reply:
I think its fine. If it turns out this is the only endpoint that implements it, we can re-address